go/vt/vtctl/reparentutil/durability_funcs.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reparentutil
+
+import (
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+	"vitess.io/vitess/go/vt/vtctl/reparentutil/promotionrule"
+)
+
+// SemiSyncAckersForPrimary returns the list of tablets which are capable of sending Semi-Sync Acks for the given primary tablet
+func SemiSyncAckersForPrimary(primary *topodatapb.Tablet, allTablets []*topodatapb.Tablet) (semiSyncAckers []*topodatapb.Tablet) {
+	for _, tablet := range allTablets {
+		if topoproto.TabletAliasEqual(primary.Alias, tablet.Alias) {
+			continue
+		}
+		if IsReplicaSemiSync(primary, tablet) {
+			semiSyncAckers = append(semiSyncAckers, tablet)
+		}
+	}
+	return
+}
+
+// RevokeForTablet checks whether we have reached enough tablets such that the given primary eligible tablet cannot accept any new writes
+func RevokeForTablet(primaryEligible *topodatapb.Tablet, tabletsReached []*topodatapb.Tablet, allTablets []*topodatapb.Tablet) bool {
```

Comment:
for these functions that check and return bools, i'd suggest renaming to from `<Thing>` to `Is<Thing>` to make them read a little nicer where they are used

Reply:
How do you feel about, `haveRevokedForTablet`? instead of `isRevokedForTablet`?

Reply:
I have made the change for `haveRevokedForTablet`. If you feel the other name is better, I'll rename it

Reply:
fine by me! as long as it implies the boolean/question nature, and we're consistent with what we use for both, i'm happy :)